src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers_UnsupportedCodeActions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.UnifiedSuggestions;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions
+{
+    internal static partial class CodeActionHelpers
+    {
+        private static readonly ImmutableArray<string> s_notSupportFixerAndRefactoringProviderNames = ImmutableArray.Create(
+            // Extract method requires inline rename support, not supported by lsp.
+            PredefinedCodeRefactoringProviderNames.ExtractMethod,
+            // There is no good way to modify the project file in lsp server.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1810443
+            PredefinedCodeRefactoringProviderNames.EnableNullable);
```

Comment:
Rather than being on an allow list, shouldn't this code action just have the "RequiresNonDocumentChange" tag, since it requires a non-document change?

Reply:
let me check if this is the `RequiresNonDocumentChange` is designed to used in this way

Reply:
Ah yep, let me update this

Reply:
Done